Mathlib/NumberTheory/RamificationInertia.lean

```diff
@@ -203,41 +207,35 @@ theorem inertiaDeg_of_subsingleton [hp : p.IsMaximal] [hQ : Subsingleton (S ⧸
   exact dif_neg fun h => hp.ne_top <| h.symm.trans comap_top
 
 @[simp]
-theorem inertiaDeg_algebraMap [Algebra R S] [Algebra (R ⧸ p) (S ⧸ P)]
-    [IsScalarTower R (R ⧸ p) (S ⧸ P)] [hp : p.IsMaximal] :
+theorem inertiaDeg_algebraMap [Algebra R S] [P.LiesOver p] [p.IsMaximal] :
     inertiaDeg (algebraMap R S) p P = finrank (R ⧸ p) (S ⧸ P) := by
   nontriviality S ⧸ P using inertiaDeg_of_subsingleton, finrank_zero_of_subsingleton
-  have := comap_eq_of_scalar_tower_quotient (algebraMap (R ⧸ p) (S ⧸ P)).injective
-  rw [inertiaDeg, dif_pos this]
-  congr
-  refine Algebra.algebra_ext _ _ fun x' => Quotient.inductionOn' x' fun x => ?_
-  change Ideal.Quotient.lift p _ _ (Ideal.Quotient.mk p x) = algebraMap _ _ (Ideal.Quotient.mk p x)
-  rw [Ideal.Quotient.lift_mk, ← Ideal.Quotient.algebraMap_eq P, ← IsScalarTower.algebraMap_eq,
-    ← Ideal.Quotient.algebraMap_eq, ← IsScalarTower.algebraMap_apply]
+  rw [inertiaDeg, dif_pos (over_def P p).symm]
+
+theorem inertiaDeg_pos [p.IsMaximal] [Nontrivial (S ⧸ P)] [Algebra R S] [Module.Finite R S]
```

Comment:
I don't think `[Nontrivial (S ⧸ P)]` is nessecary? `S / P` is trivial should imply `P` is the unit ideal and thus `p` must also be the unit ideal, which cannot happen.